dojo/tools/qualys/csv_parser.py

```diff
@@ -0,0 +1,194 @@
+import csv
+import io
+import logging
+import re
+from datetime import datetime
+
+from dojo.models import Endpoint, Finding
+
+_logger = logging.getLogger(__name__)
+
+
+def parse_csv(csv_file) -> [Finding]:
+    """
+    Parses Qualys Report in CSV format
+    Args:
+        csv_file:
+    Returns:
+    """
+
+    default_keys = [
+        'IP',
+        'Network',
+        'DNS',
+        'NetBIOS',
+        'Tracking Method',
+        'OS',
+        'IP Status',
+        'QID',
+        'Title',
+        'Vuln Status',
+        'Type',
+        'Severity',
+        'Port',
+        'Protocol',
+        'FQDN',
+        'SSL',
+        'First Detected',
+        'Last Detected',
+        'Times Detected',
+        'Date Last Fixed',
+        'CVE ID',
+        'Vendor Reference',
+        'Bugtraq ID',
+        'CVSS3',
+        'CVSS3 Base',
+        'CVSS3 Temporal',
+        'Threat',
+        'Impact',
+        'Solution',
+        'Exploitability',
+        'Associated Malware',
+        'PCI Vuln',
+        'Ticket State',
+        'Instance',
+        'OS CPE',
+        'Category',
+        'Associated Tags']
+
+    content = csv_file.read()
+    if isinstance(content, bytes):
+        content = content.decode('utf-8')
+    csv_reader = csv.DictReader(
+        io.StringIO(content),
+        delimiter=',',
+        quotechar='"',
+        fieldnames=default_keys)
+
+    report_findings = get_report_findings(csv_reader)
+    dojo_findings = build_findings_from_dict(report_findings)
+
+    return dojo_findings
+
+
+def get_report_findings(csv_reader) -> [dict]:
+    """
+    Filters out the unneeded information at the beginning of the Qualys CSV report.
+    Args:
+        csv_reader:
+
+    Returns:
+
+    """
+
+    report_findings = []
+
+    for row in csv_reader:
+        if row.get('Title') and row['Title'] != 'Title':
+            report_findings.append(row)
+
+    return report_findings
+
+
+def get_references(cve_list):
```

Comment:
There is a new mechanism for vulnerability ids like CVEs or GHSA's. You can fill them in a list called finding.unsaved_vulnerability_ids. They will all be showed in the UI and will all be clickable to show the content. I would prefer using this than building your own set of references.